hazelcast/test/src/HazelcastTests2.cpp

```diff
@@ -1743,6 +1657,282 @@ namespace hazelcast {
+                ASSERT_EQ(value, (*map_->get<std::string, byte>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ("127", response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_int16_t) {
+                int16_t value = INT16_MIN;
+                map_->set("key", value).get();
+                ASSERT_EQ(value, (*map_->get<std::string, int16_t>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ(std::to_string(value), response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_int32_t) {
+                int32_t value = INT32_MAX;
+                map_->set("key", value).get();
+                ASSERT_EQ(value, (*map_->get<std::string, int32_t>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ(std::to_string(value), response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_int64_t) {
+                int64_t value = INT64_MIN;
+                map_->set("key", value).get();
+                ASSERT_EQ(value, (*map_->get<std::string, int64_t>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ(std::to_string(value), response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_double) {
+                double value = 123.6;
+                map_->set("key", value).get();
+                ASSERT_EQ(value, (*map_->get<std::string, double>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ("123.6", response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_string) {
+                std::string value = "string_value";
+                map_->set("key", value).get();
+                ASSERT_EQ(value, (*map_->get<std::string, std::string>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ(value, response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_utf_string) {
+                std::string value = u8"Iñtërnâtiônàlizætiøn";
+                map_->set("key", value).get();
+                ASSERT_EQ(value, (*map_->get<std::string, std::string>("key").get()));
+                auto response = get_value_from_server();
+                ASSERT_TRUE(response.success);
+                ASSERT_EQ(value, response.result);
+            }
+
+            TEST_F(SerializationWithServer, test_emoji) {
+                std::string value = u8"1⚐中\\uD83D\\uDCA62\\uD83D\\uDE2D\\u200D\\uD83D\\uDE46\\uD83D\\uDE145";
```

Comment:
Are you aware of the fact that you are not sending the encoding of the emojis but instead sending the Java escaped representations of them?

With the double backslashes (`\\`) escaped the string literally becomes: `\uD83D\uDCA62\uD83D\uDE2D\u200D\uD83D\uDE46\uD83D\uDE145`, so as far as the client and the server is concerned, this is just a string of ASCII characters, not emojis. 

For example, the correct escaping of the emoji 💦 in C++ should be like this: https://godbolt.org/z/Kd317v.

Both C++ and Java have `\uNNNN` for UTF character escaping but the meaning is different. See https://en.cppreference.com/w/cpp/language/escape.

Reply:
Changed.